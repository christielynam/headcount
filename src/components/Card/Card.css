.card {
  width: 200px;
  height: 300px;
  margin: 0 5px 10px 5px;
  border: 1px solid #393939;
  text-align: center;
  cursor: pointer;
}

.list {
  list-style-type: none;
} 

h4 {
  color: #393939;
}

.low {
  color: aqua
}

.high {
  color: blue
}

.active {
  border: 2px solid red
}