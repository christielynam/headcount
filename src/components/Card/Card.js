import React from 'react'
import './Card.css'

const Card = ({ location, stats, active, toggleActive, comparison }) => {
  const displayStats = Object.entries(stats).map(year => (
    <li 
      key={year[0]}
      className={year[1] < 0.5 ? 'low' : 'high'}
    >
      {year[0]}: {year[1]}
    </li>
  ))

  return (
    <div className={active ? 'active card' : 'card'} onClick={() => toggleActive(location)}>
      <h4>{location}</h4>
      <ul className='list'>
        {displayStats}
      </ul>
    </div>
  )

}

export default Card