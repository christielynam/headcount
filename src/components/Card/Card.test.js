import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme'

describe('Card', () => {
  let wrapper
  let mockLocation
  let mockStats
  let mockToggleActive

  beforeEach(() => {
    mockLocation = 'COLORADO'
    mockStats = {2004: '1', 2005: '1', 2006: '1', 2007: '1'}
    mockToggleActive = jest.fn()
    wrapper = shallow(<Card location={mockLocation} active={false} stats={mockStats} toggleActive={mockToggleActive} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call toggleActive when a card is clicked', () => {
    const card = wrapper.find('.card')

    card.simulate('click')

    expect(mockToggleActive).toHaveBeenCalled()
  })
})