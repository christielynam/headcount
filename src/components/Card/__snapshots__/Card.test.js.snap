// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    active={false}
    location="COLORADO"
    stats={
      Object {
        "2004": "1",
        "2005": "1",
        "2006": "1",
        "2007": "1",
      }
    }
    toggleActive={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          COLORADO
        </h4>,
        <ul
          className="list"
        >
          <li
            className="high"
          >
            2004
            : 
            1
          </li>
          <li
            className="high"
          >
            2005
            : 
            1
          </li>
          <li
            className="high"
          >
            2006
            : 
            1
          </li>
          <li
            className="high"
          >
            2007
            : 
            1
          </li>
        </ul>,
      ],
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="high"
            >
              2004
              : 
              1
            </li>,
            <li
              className="high"
            >
              2005
              : 
              1
            </li>,
            <li
              className="high"
            >
              2006
              : 
              1
            </li>,
            <li
              className="high"
            >
              2007
              : 
              1
            </li>,
          ],
          "className": "list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "2004",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2004",
                ": ",
                "1",
              ],
              "className": "high",
            },
            "ref": null,
            "rendered": Array [
              "2004",
              ": ",
              "1",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2005",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2005",
                ": ",
                "1",
              ],
              "className": "high",
            },
            "ref": null,
            "rendered": Array [
              "2005",
              ": ",
              "1",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2006",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2006",
                ": ",
                "1",
              ],
              "className": "high",
            },
            "ref": null,
            "rendered": Array [
              "2006",
              ": ",
              "1",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2007",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2007",
                ": ",
                "1",
              ],
              "className": "high",
            },
            "ref": null,
            "rendered": Array [
              "2007",
              ": ",
              "1",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            COLORADO
          </h4>,
          <ul
            className="list"
          >
            <li
              className="high"
            >
              2004
              : 
              1
            </li>
            <li
              className="high"
            >
              2005
              : 
              1
            </li>
            <li
              className="high"
            >
              2006
              : 
              1
            </li>
            <li
              className="high"
            >
              2007
              : 
              1
            </li>
          </ul>,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="high"
              >
                2004
                : 
                1
              </li>,
              <li
                className="high"
              >
                2005
                : 
                1
              </li>,
              <li
                className="high"
              >
                2006
                : 
                1
              </li>,
              <li
                className="high"
              >
                2007
                : 
                1
              </li>,
            ],
            "className": "list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "2004",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2004",
                  ": ",
                  "1",
                ],
                "className": "high",
              },
              "ref": null,
              "rendered": Array [
                "2004",
                ": ",
                "1",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2005",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2005",
                  ": ",
                  "1",
                ],
                "className": "high",
              },
              "ref": null,
              "rendered": Array [
                "2005",
                ": ",
                "1",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2006",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2006",
                  ": ",
                  "1",
                ],
                "className": "high",
              },
              "ref": null,
              "rendered": Array [
                "2006",
                ": ",
                "1",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2007",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2007",
                  ": ",
                  "1",
                ],
                "className": "high",
              },
              "ref": null,
              "rendered": Array [
                "2007",
                ": ",
                "1",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
